app/bundles/PageBundle/Controller/PageController.php

```diff
@@ -547,6 +541,18 @@ public function editAction(
                     // form is valid so process the data
                     $model->saveEntity($entity, $this->getFormButton($form, ['buttons', 'save'])->isClicked());
 
+                    if ($pageConfig->isDraftEnabled() && !empty($entity->getId())) {
+                        $this->dispatcher->dispatch(new PageEditSubmitEvent(
+                            $existingPage,
+                            $entity,
+                            $this->getFormButton($form, ['buttons', 'save'])->isClicked(),
+                            $this->getFormButton($form, ['buttons', 'apply'])->isClicked(),
+                            $form->get('buttons')->has('save_draft') && $this->getFormButton($form, ['buttons', 'save_draft'])->isClicked(),
+                            $form->get('buttons')->has('apply_draft') && $this->getFormButton($form, ['buttons', 'apply_draft'])->isClicked(),
+                            $form->get('buttons')->has('discard_draft') && $this->getFormButton($form, ['buttons', 'discard_draft'])->isClicked()
+                        ));
+                    }
```

Comment:
```suggestion
                    if ($pageConfig->isDraftEnabled() && !empty($entity->getId())) {
                        $this->dispatcher->dispatch(new PageEditSubmitEvent(
                            $existingPage,
                            $entity,
                            $this->isButtonClicked($form, 'save'),
                            $this->isButtonClicked($form, 'apply'),
                            $this->isButtonClicked($form, 'save_draft'),
                            $this->isButtonClicked($form, 'apply_draft'),
                            $this->isButtonClicked($form, 'discard_draft'),
                        ));
                    }
```
Let's reduce the duplication and improve readability here as well.